### DISCUSSION TITLE
python 3.13 and modelx

### DISCUSSION THREAD

**Discussion by bakerwy** [fumitoh/modelx#163] [Q&A]
As python 3.13 has been released. It seems there are quite huge changes in this version especially GIL release(free threading) and JIT (just in time compilation). 

Is there any methods modelx can utilize these features or future plans modelx for these? (Especially free threading)

**Comment by fumitoh** [comment]
Free-threading is one of the most exciting features that has been added to Python, althoug it's not enabled by default.
I definately want modelx to support free-threading. It would be like `Projection[1]`, `Projection[2]`, `Projection[3]`, `Projection[4]` all running on different threads.
My understanding of JIT is it's managed by Python itself, so there's nothing left to do on modelx' side.

**Comment by fumitoh** [comment]
Has anyone had a chance to explore Python 3.13 with free-threading?

https://py-free-threading.github.io/

I tried the original development version a coupule years ago, but haven't touched it since then.   How do you install free-threaded Python 3.13 with numpy, pandas and all essentioal libraries?

**Reply by bakerwy** [reply]
I tried some basic examples and it works. but modelx / numpy / pandas in 3.13t and it shows errors and not working. I guess it is not automatically installed...

https://py-free-threading.github.io/porting/#__tabbed_1_1

**Reply by bakerwy** [reply]
https://github.com/numpy/numpy/issues/26157

seeing this, Linux and Mac built. 

3.13t wheel built very recently. I haven't tried yet.

**Reply by fumitoh** [reply]
Thanks, I guess I should wait until more information and libraries get available for the free-threading build.
